docs/src/content/docs/utilities/watch-directory.mdx

```diff
@@ -0,0 +1,36 @@
+---
+title: watchDirectory
+description: In development, will reload the Astro dev server if any files within the directory has changed.
+---
+
+In development, `watchDirectory` will reload the Astro dev server if any files within
+the directory has changed. For example:
+
+```ts title="my-integration/index.ts" {4,14}
+import {
+	defineIntegration,
+	createResolver,
+	watchDirectory
+} from "astro-integration-kit";
+
+export default defineIntegration({
+	// ...
+	setup() {
+		const { resolve } = createResolver(import.meta.url)
+
+		return {
+			"astro:config:setup": (params) => {
+				watchDirectory(params, resolve())
+			}
+		}
+	}
+})
+```
+
+## Limitations
+
+Those are due to limitations in Astro core:
+
+1. This utility only watches files (as deep as possible) under the specified directory
+2. Directories under the specified directory are not taken into account
+3. Additions and deletions are not taken into account
```

Comment:
I think deletions are considered. The point is that we call `addWatchFile` for each file, so any file that exists is watched and it should consider the file being deleted, but files that are added were not present when calling this function so they are not watched.

Reply:
Would you mind writing a suggestion?

Reply:
```suggestion
3. Files added to the directory won't be detected until one of the old files is changed
```